src/service/chain_rewardable_entities.proto

```diff
@@ -0,0 +1,177 @@
+// TODO: Design for different key formats - track "archetype" of key ... helium
+// vs. solana, network prefixes etc.
+// TODO: Consider OTHER Rewardable Entities (REs) in the future
+// TODO: Signature STUFF in req's and reports
+
+syntax = "proto3";
+
+package helium.chain_rewardable_entities;
+
+enum pub_key_format {
+  pub_key_format_UNKNOWN = 0; // Unknown format, should not be used
+  pub_key_format_B58 = 1;     // Base58 format
+  pub_key_format_UTF8 = 2;    // UTF8 format
+}
+
+message pub_key {
+  string value = 1;
+  pub_key_format format = 2;
+}
+
+// ------------------------------------------------------------------
+// Helium Hotspot "rewardable entities" and their metadata
+// ------------------------------------------------------------------
+enum mobile_hotspot_device_type {
+  mobile_hotspot_device_type_UNKNOWN =
+      0; // Unknown device type, should not be used
+  mobile_hotspot_device_type_WIFI_INDOOR = 1;
+  mobile_hotspot_device_type_WIFI_OUTDOOR = 2;
+  mobile_hotspot_device_type_WIFI_DATA_ONLY = 3;
+}
+
+message mobile_hotspot_metadata {
+  // Timestamp when the NFT was created
+  uint64 created_at_ms = 1;
+  // Serial number of the mobile hotspot device (can be null or empty... in the
+  // case of data-only devices)
+  optional string serial_number = 2;
+  // Type of the mobile hotspot device
+  mobile_hotspot_device_type device_type = 3;
+  // The asserted location in hex format
+  string asserted_hex = 4;
```

Comment:
Do we not want the other stuff from mobile hotspot info? Elevation? Device tilt? Etc?

Reply:
MVP use cases don't need it - but could well be useful to others. Hopefully others can offer some perspective here - @madninja or maybe even someone from HG